src/extractor/ConfigurationExtractor.ts

```diff
@@ -36,9 +36,24 @@ export class ConfigurationExtractor {
       (ConfigurationExtractor.inferCanonical(resources) || 'http://example.org');
     if (!fhirVersion?.length) {
       const fhirVersionFromResources = ConfigurationExtractor.inferString(resources, 'fhirVersion');
-      fhirVersion = utils.isSupportedFHIRVersion(fhirVersionFromResources)
-        ? [fhirVersionFromResources]
-        : ['4.0.1'];
+      if (utils.isSupportedFHIRVersion(fhirVersionFromResources)) {
+        fhirVersion = [fhirVersionFromResources];
+      }
+    }
+    if (specifiedFHIRVersion) {
+      // If there is a detected version, check if it matches the specified version
+      if (fhirVersion?.length && fhirVersion.indexOf(specifiedFHIRVersion) === -1) {
+        logger.warn(
+          `FHIR Version mismatch error: specified version is: ${specifiedFHIRVersion} while detected version is: ${fhirVersion.join(
```

Comment:
Since this is a warning now, does it make sense to update the text here to say "FHIR Version mismatch warning: ..." instead?

Reply:
I agree here too.  Good catch, @jafeltra!

Reply:
Done.